### PR TITLE
gateio cross_margin and crossMargin in accountByType

### PR DESCRIPTION
**Pull request by samgermain**
This is how you transfer to cross margin

### CODE REVIEW

**Comment by samgermain** [comment]
@kroitor This has nothing to do with a type of "margin", gateio requires you to set the transfer account as `cross_margin` to transfer in and out of the `cross_margin` account, without these values it's currently not possible to transfer into those accounts

**Comment by samgermain** [comment]
@kroitor This is why it's needed, the collateral in `cross_margin` on `gateio` is independent from the collateral in `margin`, right now we can't transfer in and out of the cross margin account

The examples below just show that the cross_margin balance changes, but the isolated margin balance does not

```
% gateio fetchBalance '{"marginType": "cross"}'
gateio.fetchBalance ([object Object])

{
  info: {...},
  USDT: { free: 11.502896176, used: 0, total: 11.502896176 },
  ADA: { free: 0, used: 0, total: 0 },
  free: { USDT: 11.502896176, ADA: 0 },
  used: { USDT: 0, ADA: 0 },
  total: { USDT: 11.502896176, ADA: 0 }
}

% gateio fetchBalance '{"marginType": "isolated"}'
gateio.fetchBalance ([object Object])

{
  info: [...],
  'BTC/USDT': {
    BTC: { free: 0, used: 0, total: 0 },
    USDT: { free: 2, used: 0, total: 2 },
    free: { BTC: 0, USDT: 2 },
    used: { BTC: 0, USDT: 0 },
    total: { BTC: 0, USDT: 2 }
  },
  'ADA/USDT': {
    ADA: { free: 0, used: 0, total: 0 },
    USDT: { free: 19.7739719088, used: 0, total: 19.7739719088 },
    free: { ADA: 0, USDT: 19.7739719088 },
    used: { ADA: 0, USDT: 0 },
    total: { ADA: 0, USDT: 19.7739719088 }
  }
}

% gateio transfer USDT 1 cross_margin spot        
gateio.transfer (USDT, 1, cross_margin, spot)

{
  id: undefined,
  timestamp: 1651272216760,
  datetime: '2022-04-29T22:43:36.760Z',
  currency: 'USDT',
  amount: 1,
  fromAccount: 'cross_margin',
  toAccount: 'spot',
  status: undefined,
  info: ''
}

% gateio fetchBalance '{"marginType": "cross"}'
gateio.fetchBalance ([object Object])

{
  info: {...},
  USDT: { free: 10.502896176, used: 0, total: 10.502896176 },
  ADA: { free: 0, used: 0, total: 0 },
  free: { USDT: 10.502896176, ADA: 0 },
  used: { USDT: 0, ADA: 0 },
  total: { USDT: 10.502896176, ADA: 0 }
}
```

```
% gateio fetchBalance '{"marginType": "isolated"}'
Debugger listening on ws://127.0.0.1:54011/f6c47c01-a0bf-4b91-b1bf-e51413fd415b
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
2022-04-29T22:48:14.720Z
Node.js: v14.17.0
CCXT v1.81.9
gateio.fetchBalance ([object Object])
2022-04-29T22:48:16.652Z iteration 0 passed in 209 ms

{
  info: [...],
  'BTC/USDT': {
    BTC: { free: 0, used: 0, total: 0 },
    USDT: { free: 2, used: 0, total: 2 },
    free: { BTC: 0, USDT: 2 },
    used: { BTC: 0, USDT: 0 },
    total: { BTC: 0, USDT: 2 }
  },
  'ADA/USDT': {
    ADA: { free: 0, used: 0, total: 0 },
    USDT: { free: 19.7739719088, used: 0, total: 19.7739719088 },
    free: { ADA: 0, USDT: 19.7739719088 },
    used: { ADA: 0, USDT: 0 },
    total: { ADA: 0, USDT: 19.7739719088 }
  }
}
```

File: js/gateio.js [hunk 1]
```diff
@@ -408,7 +408,8 @@ module.exports = class gateio extends Exchange {
                     'spot': 'spot',
                     'margin': 'margin',
                     'cross_margin': 'cross_margin',
-                    'crossMargin': 'cross_margin',
+                    'cross': 'cross_margin',
+                    'inverse': 'margin',
```
Comment (samgermain) [inline js/gateio.js:412]: Why would inverse be margin?

File: js/gateio.js [hunk 2]
```diff
@@ -408,7 +408,8 @@ module.exports = class gateio extends Exchange {
                     'spot': 'spot',
                     'margin': 'margin',
                     'cross_margin': 'cross_margin',
-                    'crossMargin': 'cross_margin',
```
Comment (samgermain) [inline js/gateio.js]: The idea behind this was consistency, because everything else we have is camelcase

File: js/gateio.js (hunk 1 above)
Comment (kroitor) [reply js/gateio.js:412]: I meant `isolated`, will fix.

File: js/gateio.js (hunk 2 above)
Comment (kroitor) [reply js/gateio.js]: Not for exchange-specific account names, though. It's very hard to guess `'crossMargin'` without looking into the code. All others could be guesses by looking at either CCXT docs or the exchange API docs.

**Pull request merged**